ios/brave-ios/Sources/BraveWallet/Crypto/WalletActions/DepositTokenView.swift

```diff
@@ -28,9 +28,15 @@ struct DepositTokenView: View {
   @State private var isPresentingAddAccount: Bool = false
   @State private var isPresentingAddAccountConfirmation: Bool = false
 
-  private var availableAccounts: [BraveWallet.AccountInfo] {
-    guard let token = depositTokenStore.prefilledToken else { return [] }
-    return depositTokenStore.allAccounts.filter({ $0.coin == token.coin })
+  private func availableAccounts(
+    for token: BraveWallet.BlockchainToken
+  ) -> [BraveWallet.AccountInfo] {
+    guard
+      let tokenNetwork = depositTokenStore.allNetworks.first(where: {
+        $0.chainId == token.chainId
+      })
+    else { return [] }
+    return depositTokenStore.allAccounts.accountsFor(network: tokenNetwork)
```

Comment:
i think it's safe by just checking `chainId` since mainnet and testnet has its own unique chainId meaning the `guard` will guarantee to return the correct NetworkInfo.
the extension of `BraveWallet.KeyringId` is essentially just checking the chainId as well.

Reply:
> i think it's safe by just checking chainId since mainnet and testnet has its own unique chainId meaning the guard will guarantee to return the correct NetworkInfo.

One less spot to update (or miss) should this change in the future though.